packages/frontend/src/features/sqlRunner/components/SqlEditor.tsx

```diff
@@ -258,18 +258,22 @@ export const SqlEditor: FC<{
     const decorationsCollectionRef =
         useRef<editor.IEditorDecorationsCollection | null>(null); // Ref to store the decorations collection
 
-    const onMount: OnMount = useCallback((editorObj, monacoObj) => {
-        editorRef.current = editorObj;
-        decorationsCollectionRef.current =
-            editorObj.createDecorationsCollection(); // Initialize the decorations collection
-        editorObj.addCommand(
-            monacoObj.KeyMod.CtrlCmd | monacoObj.KeyCode.Enter,
-            () => {
-                // When the editor is mounted, the onSubmit callback should be set to the latest value, otherwise it will be set to the initial value on the first render
-                onSubmitRef.current?.();
-            },
-        );
-    }, []);
+    const onMount: OnMount = useCallback(
+        (editorObj, monacoObj) => {
+            editorRef.current = editorObj;
+            decorationsCollectionRef.current =
+                editorObj.createDecorationsCollection();
+            editorObj.addCommand(
+                monacoObj.KeyMod.CtrlCmd | monacoObj.KeyCode.Enter,
+                () => {
+                    const currentSql = editorObj.getValue();
+                    if (!onSubmit) return;
+                    onSubmit(currentSql ? currentSql : '');
```

Comment:
```suggestion
                   onSubmit(currentSql ?? '');
```
Can be similar to what you did in `ContentPanel`